apps/api/src/domain/models/media-item/commands/publish-media-item.command.integration.spec.ts

```diff
@@ -0,0 +1,193 @@
+import { assertCommandSuccess } from '../../__tests__/command-helpers/assert-command-success';
+import { CommandAssertionDependencies } from '../../__tests__/command-helpers/types/CommandAssertionDependencies';
+import buildDummyUuid from '../../__tests__/utilities/buildDummyUuid';
+import { MediaItem } from '../entities/media-item.entity';
+import { PublishMediaItem } from './publish-media-item.command';
+import { PublishMediaItemCommandHandler } from './publish-media-item.command-handler';
+
+const commandType = 'PUBLISH_MEDIA_ITEM';
+
+const dummyUUID = buildDummyUuid();
+
+const validCommandFSA: FluxStandardAction<DTO<PublishMediaItem>> = {
+    type: commandType,
+    payload: { id: dummyUUID },
+};
+
+const buildInvalidCommandFSA = (id: unknown) => ({
+    type: commandType,
+    payload: { id },
+});
+
+const unpublishedMediaItem = getValidResourceInstanceForTest(ResourceType.mediaItem).clone({
+    published: false,
+    id: dummyUUID,
+});
+
+const buildFullSnapshot = (preExistingMediaItem: MediaItem) =>
+    buildInMemorySnapshot({
+        resources: {
+            mediaItem: [preExistingMediaItem],
+        },
+    });
+
+const initialState = buildFullSnapshot(unpublishedMediaItem);
+
+describe('PublishMediaItem', () => {
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let arangoConnectionProvider: ArangoConnectionProvider;
+
+    let idManager: IIdManager;
+
+    let assertionHelperDependencies: Omit<CommandAssertionDependencies, 'idManager'>;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, arangoConnectionProvider } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: 'testonly-333',
+                // generateRandomTestDatabaseName(),
+            }));
+
+        commandHandlerService.registerHandler(
+            commandType,
+            new PublishMediaItemCommandHandler(
+                testRepositoryProvider,
+                idManager,
+                ResourceType.mediaItem
+            )
+        );
+
+        assertionHelperDependencies = {
+            testRepositoryProvider,
+            commandHandlerService,
+        };
+    });
+
+    afterAll(async () => {
+        await arangoConnectionProvider.dropDatabaseIfExists();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe('when the command is valid', () => {
+        it('should succeed', async () => {
+            await assertCommandSuccess(assertionHelperDependencies, {
+                buildValidCommandFSA: () => validCommandFSA,
+                initialState,
+                checkStateOnSuccess: async () => {
+                    const result = await testRepositoryProvider
+                        .forResource(ResourceType.mediaItem)
```

Comment:
We could store a reference to the `resourceType` and to the `mediaItemRepository` at the top level to make things less verbose.